domain/schema/model.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package schema
+
+import (
+	"github.com/juju/juju/core/database"
+)
+
+// ModelDDL is used to create the schema used for each model.
+func ModelDDL() []database.Delta {
+	schemas := []func() database.Delta{
+		modelChangeLogSchema,
+		modelConfigSchema,
+	}
+
+	deltas := make([]database.Delta, 0, len(schemas))
+	for _, fn := range schemas {
+		deltas = append(deltas, fn())
+	}
+
+	return deltas
+}
+
+func modelChangeLogSchema() database.Delta {
+	return database.MakeDelta(`
+CREATE TABLE change_log_edit_type (
```

Comment:
Let's not duplicate all this here - isn't it exactly the same as for the controller change log schema?
Should we use a helper function to generate it for both?

Reply:
Let's do this. The trigger set-ups are all boilerplatey too - we could use a helper for those.